packages/app-webdir-ui/src/helpers/search.js

```diff
@@ -21,21 +21,33 @@ export const engineNames = {
 };
 
 export function logClick(query, docId, reqId, tags, { ...props }) {
-  function sendData(resolve, reject) {
+  async function sendData(resolve, reject) {
     const data = {
       query,
       doc_id: docId,
       req_id: reqId,
       tags,
     };
-    axios
-      .post(`${props.API_URL}${props.searchApiVersion}webdir-click`, data)
-      .then(res => {
-        resolve(res);
-      })
-      .catch(err => {
-        reject(err);
-      });
+    let response;
+    if (props.loggedIn) {
+      const tokenResponse = await axios.get(`${props.API_URL}/session/token`);
+      const headers = {
+        "X-CSRF-Token": tokenResponse.data,
+      };
+      response = await axios.post(
+        `${props.API_URL}${props.searchApiVersion}webdir-click`,
```

Comment:
Regarding building a url, at first glance these 2 lines look like an error
`${ props.API_URL }/session/token`
`${ props.API_URL }${ props.searchApiVersion }webdir-click`,

After I looked closer I was able to see the `searchApiVersion` contains a `/`
if we change searchApiVersion from `/api/v1/` to `/api/v1` then the following 2 lines look more consistent
`${ props.API_URL }/session/token`
`${ props.API_URL }${ props.searchApiVersion }/webdir-click`,

Just something to consider, no change required if we think this is not a huge concern. 
Could users technically add these props without proper `/`?
We just don't want the string to endup without any `/` joining the parts

Reply:
Yup, documented in https://github.com/ASU/asu-unity-stack/pull/820#discussion_r933697402